tron/serialize/runstate/dynamodb_state_store.py

```diff
@@ -106,7 +106,7 @@ def save(self, key_value_pairs) -> None:
                     log.info(f"save queue size {qlen} > {MAX_SAVE_QUEUE}, sleeping 5s")
                     time.sleep(5)
                     continue
-                self.save_queue[key] = pickle.dumps(val)
+                self.save_queue[key] = val
```

Comment:
just realized this, i don't think ordereddict is thread-safe like dequeue was. we may need to add locking here and when popping.

Reply:
ah, good point. here's a PR for that: https://github.com/Yelp/Tron/pull/760